### DISCUSSION TITLE
Error on GNS3 with Mac M2 Chip

### DISCUSSION THREAD

**Discussion by Christopherhausner** [GNS3/gns3-gui#3567] [General]
I get this error when trying to connect 

Error while saving settings: GNS3VM: Error while executing VMware command: vmrun has returned an error: unknown

Try to run '/Applications/VMware Fusion.app/Contents/Library/vmrun' -T fusion checkToolsState '/Users/chrishausner/Virtual Machines.localized/GNS3 ARM.vmwarevm/GNS3 ARM.vmx' in a terminal to see more details.
And make sure GNS3 and VMware run under the same user.

I went in and I can find the files they mention. Any ideas?

**Comment by eantowne** [comment]
If "/Applications/VMware Fusion.app/Contents/Library/vmrun" does not exist, this sounds like an issue with WMware, not GNS3.

From a terminal, what is the output of: ls -al /Applications/VMware Fusion.app/Contents/Library/

Also as a side note, GNS3 is not really supported on Apple Silicon. Your mileage may vary.

**Reply by Christopherhausner** [reply]
chrishausner@Chriss-MacBook-Pro ~ % ls -al /Applications/VMware Fusion.app/Contents/Library/
ls: /Applications/VMware: No such file or directory
ls: Fusion.app/Contents/Library/: No such file or directory
chrishausner@Chriss-MacBook-Pro ~ % ls -al '/Applications/VMware Fusion.app/Contents/Library/'
total 551624
drwxr-xr-x  65 root  wheel      2080 Feb 29 14:36 .
drwxr-xr-x  11 root  wheel       352 Feb 29 14:36 ..
-rwxr-xr-x@  1 root  wheel     10746 Feb 12 20:50 Create Mavericks Installer.tool
-rwxr-xr-x@  1 root  wheel     10981 Feb 12 20:50 Create macOS Installer.tool
drwxr-xr-x   3 root  wheel        96 Feb 12 20:50 Deploy VMware Fusion.mpkg
-rwxr-xr-x@  1 root  wheel      1862 Feb 12 20:50 Initialize VMware Fusion.tool
drwxr-xr-x  12 root  wheel       384 Feb 29 14:36 LaunchServices
drwxr-xr-x   3 root  wheel        96 Feb 12 20:50 QuickLook
drwxr-xr-x   3 root  wheel        96 Oct 23  2020 VMware Fusion Applications Menu.app
-rwxr-xr-x@  1 root  wheel     19960 Feb 12 20:50 VMware Fusion Problem Reporter.tool
drwxr-xr-x   3 root  wheel        96 Feb 12 20:50 VMware Fusion Start Menu.app
drwxr-xr-x  12 root  wheel       384 Feb 12 20:53 VMware OVF Tool
-rwxr-xr-x   1 root  wheel   3592624 Feb 12 20:53 amsrv
drwxr-xr-x   7 root  wheel       224 Feb 12 20:53 dockers
drwxr-xr-x   3 root  wheel        96 Feb 12 20:50 icu
drwxr-xr-x   4 root  wheel       128 Feb 12 20:50 isoimages
drwxr-xr-x  14 root  wheel       448 Feb 12 20:50 languageSpecificMacToWinKeymap
drwxr-xr-x   4 root  wheel       128 Feb 12 20:53 licenses
drwxr-xr-x   8 root  wheel       256 Feb 12 20:50 messages
-rwxr-xr-x   1 root  wheel    470064 Feb 12 20:53 mkisofs
-rwxr-xr-x   1 root  wheel    321888 Feb 12 20:53 mkntfs
-rwxr-xr-x   1 root  wheel   9526688 Feb 12 20:53 mksSandbox
-rwxr-xr-x   1 root  wheel  10729680 Feb 12 20:53 mksSandbox-debug
-rwxr-xr-x   1 root  wheel  10169792 Feb 12 20:53 mksSandbox-stats
-rwxr-xr-x   1 root  wheel    187040 Feb 12 20:53 ntfscat
-rwxr-xr-x   1 root  wheel    188000 Feb 12 20:53 ntfscp
-rwxr-xr-x   1 root  wheel    187312 Feb 12 20:53 ntfsls
-rwxr-xr-x   1 root  wheel    288704 Feb 12 20:53 ntfsresize
-rw-r--r--@  1 root  wheel        21 Feb 12 20:50 productInfo
-rwxr-xr-x   1 root  wheel    234992 Feb 12 20:53 regval
-rwxr-xr-x@  1 root  wheel       465 Feb 12 20:49 relaunch.sh
drwxr-xr-x   4 root  wheel       128 Feb 12 20:49 roms
-rwxr-xr-x   1 root  wheel    907552 Feb 12 20:53 scriptreg
drwxr-xr-x  10 root  wheel       320 Feb 29 14:36 services
-rw-r--r--   1 root  wheel   1421232 Feb 12 20:50 swagger.zip
drwxr-xr-x   8 root  wheel       256 Feb 12 20:50 tools-upgraders
-rwxr-xr-x   1 root  wheel  11062704 Feb 12 20:53 tpm2emu
-rwxr-xr-x   1 root  wheel    749488 Feb 12 20:53 unrar
drwxr-xr-x  11 root  wheel       352 Feb 12 20:50 vkd
-rwxr-xr-x   1 root  wheel   1702224 Feb 12 20:53 vmnet-bridge
-rwxr-xr-x   1 root  wheel   1648624 Feb 12 20:53 vmnet-cfgcli
-rwxr-xr-x   1 root  wheel   1665744 Feb 12 20:53 vmnet-cli
-rwxr-xr-x   1 root  wheel   1304512 Feb 12 20:53 vmnet-dhcpd
-rwxr-xr-x   1 root  wheel   1768832 Feb 12 20:53 vmnet-natd
-rwxr-xr-x   1 root  wheel   1612112 Feb 12 20:53 vmnet-netifup
-rwxr-xr-x   1 root  wheel   1611968 Feb 12 20:53 vmnet-sniffer
-rwxr-xr-x   1 root  wheel  14368688 Feb 12 20:53 vmrest
-rwxr-xr-x   1 root  wheel    293856 Feb 12 20:53 vmrun
-rwxr-xr-x   1 root  wheel   2209296 Feb 12 20:53 vmss2core
-rwxr-xr-x   1 root  wheel   9656048 Feb 12 20:53 vmw_esd2iso
-rwxr-xr-x   1 root  wheel    166976 Feb 12 20:53 vmware-aewp
-rwsr-xr-x   1 root  wheel  10533360 Feb 12 20:53 vmware-authd
-rwxr-xr-x   1 root  wheel  12485200 Feb 12 20:53 vmware-cloneBootCamp
-rwxr-xr-x   1 root  wheel    167088 Feb 12 20:53 vmware-id
-rwxr-xr-x   1 root  wheel    151504 Feb 12 20:53 vmware-ntfs
-rwxr-xr-x   1 root  wheel   1630384 Feb 12 20:53 vmware-rawdiskAuthTool
-rwsr-xr-x   1 root  wheel  12536368 Feb 12 20:53 vmware-rawdiskCreator
-rwxr-xr-x   1 root  wheel  11518768 Feb 12 20:53 vmware-remotemks
-rwxr-xr-x   1 root  wheel   3345952 Feb 12 20:53 vmware-usbarbitrator
-rwxr-xr-x   1 root  wheel   5626048 Feb 12 20:53 vmware-vdiskmanager
-rwxr-xr-x   1 root  wheel  12501888 Feb 12 20:53 vmware-vmdkserver
-rwsr-xr-x   1 root  wheel  39005280 Feb 12 20:53 vmware-vmx
-rwsr-xr-x   1 root  wheel  43864896 Feb 12 20:53 vmware-vmx-debug
-rwsr-xr-x   1 root  wheel  40898288 Feb 12 20:53 vmware-vmx-stats
drwxr-xr-x  17 root  wheel       544 Feb 12 20:50 vnckeymap

Sorry for the delayed response. Once I add '' around the directory it works. I think the issue is that GNS3 is trying to find it without '' around it which it sees the space in (VMware fusion) and then doesn't work. How can I get around this? or is that not the issue?

**Reply by Christopherhausner** [reply]
Any update?
